src/test/java/com/google/cloud/spanner/pgadapter/python/ITPsycopg2Test.java

```diff
@@ -367,4 +367,298 @@ public void testDeleteParameterized() throws IOException, InterruptedException {
     expectedOutput = "(0,)\n";
     assertEquals(expectedOutput, actualOutput);
   }
+
+  @Test
+  public void testInsertUsingExecuteMany() throws IOException, InterruptedException {
+    String sql =
+        "Insert into all_types("
+            + "col_bigint, "
+            + "col_bool , "
+            + "col_bytea , "
+            + "col_float8 , "
+            + "col_int , "
+            + "col_numeric , "
+            + "col_timestamptz , "
+            + "col_date , "
+            + "col_varchar )"
+            + " values("
+            + "%(bigint)s, "
+            + "%(bool)s, "
+            + "%(bytea)s, "
+            + "%(float)s, "
+            + "%(int)s, "
+            + "%(numeric)s, "
+            + "%(tstz)s, "
+            + "%(date)s, "
+            + "%(varchar)s)";
+
+    ArrayList<String> parameters = new ArrayList<>();
+    parameters.add("9");
+    parameters.add("bool");
+    parameters.add("True");
+    parameters.add("float");
+    parameters.add("89.23");
+    parameters.add("numeric");
+    parameters.add("33.546");
+    parameters.add("tstz");
+    parameters.add("2022-02-16T14:18:02+01:00");
+    parameters.add("varchar");
+    parameters.add("hello world");
+    parameters.add("date");
+    parameters.add("2022-7-1");
+    parameters.add("bigint");
+    parameters.add("10");
+    parameters.add("bytea");
+    parameters.add("alpha beta");
+    parameters.add("int");
+    parameters.add("34");
+
+    parameters.add("bool");
+    parameters.add("False");
+    parameters.add("float");
+    parameters.add("74.567");
+    parameters.add("numeric");
+    parameters.add("99.9999");
+    parameters.add("tstz");
+    parameters.add("2021-02-16T14:18:02+01:00");
+    parameters.add("varchar");
+    parameters.add("bye bye world");
+    parameters.add("date");
+    parameters.add("2021-7-1");
+    parameters.add("bigint");
+    parameters.add("15");
+    parameters.add("bytea");
+    parameters.add("gamma theta");
+    parameters.add("int");
+    parameters.add("69");
+
+    String actualOutput =
+        executeInBatch(testEnv.getPGAdapterPort(), sql, "named_execute_many", parameters);
+    String expectedOutput = "2\n";
+
+    assertEquals(expectedOutput, actualOutput);
+
+    String sql2 =
+        "Select "
+            + "col_bigint, "
+            + "col_bool , "
+            + "col_float8 , "
+            + "col_int , "
+            + "col_numeric , "
+            + "col_timestamptz , "
+            + "col_date , "
+            + "col_varchar "
+            + "from all_types where col_bigint = 10 or col_bigint = 15";
```

Comment:
nit: add `order by id` to this sql statement to be sure that we don't get any flaky failures in the future. Although it is very common that data is returned in primary key order, it is actually not guaranteed, so we could in theory receives these rows in the order (15, 10) instead of (10, 15).

Reply:
Thanks a lot for pointing this out. I've added it